### ISSUE TITLE
Grpc Cross Compilation Issue

### ISSUE THREAD

**Issue by udhai98** [grpc/grpc#35483]
### What version of gRPC and what language are you using?

grpc version 1.38.0 
protobuf version updated to 3.18.0

### What operating system (Linux, Windows,...) and version?
Linux (petalinux environment for cross compiling) 

### What runtime / compiler are you using (e.g. python version or version of gcc)
gcc 11.2.0

### What did you do?
i need to cross compile grpc for aarch64 format for that i have made some changes in grpc package,updated the protobuf version to 3.18.0 from 3.15.8 by using update_version.py script,because petalinux environment has protobuf version 3.18.0. 

### What did you expect to see?
Need to cross compile grpc for aarch64 format using petalinux environment.

### What did you see instead?

getting lot of errors like this
error: 'google::protobuf::Arena* google::protobuf::MessageLite::GetOwningArena() const' is private within this context
  306 |     if (GetOwningArena() == from.GetOwningArena()

### Anything else we should know about your project / environment?

petalinux environment

**Comment by yashykt** [comment]
The gRPC version you are using is pretty old. There might be a version incompatibility issue between grpc and protobuf that you are running into.

Also, the origin of that error seems to be protobuf and not grpc.

**Comment by yashykt** [comment]
Can you please you try with the latest 1.60 gRPC release with the 4.25.0 protobuf release

**Comment by udhai98** [comment]
Hi,i have solved the above issue with version change of protobuff in petalinux environment.

when i am integrating the cross comipiled grpc with openolt package when i am trying to compile the opent olt package i am facing some issues with grpc library 
/home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/cortexa72-cortexa53-xilinx-linux/tmp/install/lib/libgrpc.a(ssl_transport_security.cc.o): in function `peer_from_x509(x509_st*, int, tsi_peer*)':
/home/myuser/arun/grpcarm2/grpc/src/core/tsi/ssl_transport_security.cc:389: undefined reference to `OPENSSL_free'
/home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/x86_64-petalinux-linux/usr/libexec/aarch64-xilinx-linux/gcc/aarch64-xilinx-linux/11.2.0/real-ld: /home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/cortexa72-cortexa53-xilinx-linux/tmp/install/lib/libgrpc.a(ssl_transport_security.cc.o): in function `sk_GENERAL_NAME_value':
/home/myuser/arun/grpcarm2/grpc/third_party/boringssl-with-bazel/src/include/openssl/x509v3.h:208: undefined reference to `sk_value'
/home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/x86_64-petalinux-linux/usr/libexec/aarch64-xilinx-linux/gcc/aarch64-xilinx-linux/11.2.0/real-ld: /home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/cortexa72-cortexa53-xilinx-linux/tmp/install/lib/libgrpc.a(ssl_transport_security.cc.o): in function `sk_GENERAL_NAME_pop_free':
/home/myuser/arun/grpcarm2/grpc/third_party/boringssl-with-bazel/src/include/openssl/x509v3.h:208: undefined reference to `sk_pop_free_ex'
/home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/x86_64-petalinux-linux/usr/libexec/aarch64-xilinx-linux/gcc/aarch64-xilinx-linux/11.2.0/real-ld: /home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/cortexa72-cortexa53-xilinx-linux/tmp/install/lib/libgrpc.a(ssl_transport_security.cc.o): in function `add_subject_alt_names_properties_to_peer':
/home/myuser/arun/grpcarm2/grpc/src/core/tsi/ssl_transport_security.cc:382: undefined reference to `OPENSSL_free'
/home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/x86_64-petalinux-linux/usr/libexec/aarch64-xilinx-linux/gcc/aarch64-xilinx-linux/11.2.0/real-ld: /home/myuser/arun/petalinux/2022.2/xilinx-zcu102-2022.2/tools/Xilinx/toolchain/sysroots/cortexa72-cortexa53-xilinx-linux/tmp/install/lib/libgrpc.a(ssl_transport_security.cc.o): in function `sk_GENERAL_NAME_pop_free':
/home/myuser/arun/grpcarm2/grpc/third_party/boringssl-with-bazel/src/include/openssl/x509v3.h:208: undefined reference to `sk_pop_free_ex'

**Comment by SHLEE-235** [comment]
@udhai98 

Hello, I am also experiencing a similar issue to yours.
After enabling the grpc option in petalinux-config -c rootfs, I successfully completed the steps up to petalinux-build and petalinux-build --sdk. When I sourced the configuration file in the SDK directory, I confirmed that it was loading version 3.18 of protoc. The issue lies in generating the pb.cc and pb.h files using protoc. It seems that protoc cannot find grpc_cpp_plugin, preventing it from running properly. How can I resolve this issue?

I found a related discussion here: https://github.com/grpc/grpc/issues/27034, but it didn't work for me. I am leaving this question to ask how you resolved it.
I would greatly appreciate your help.

Have a happy day today!
